src/base/ClpMap.h

```diff
@@ -32,10 +32,13 @@ template <class Key, class EntryValue, size_t MemoryUsedByEV(const EntryValue *)
 class ClpMap
 {
 public:
+    /// maximum desired entry caching duration (a.k.a. TTL), in seconds
+    using Ttl = int;
```

Comment:
> Why "using" instead of typedef?

Personally, I find type alias declarations more readable (many developers share that view) and a bit more "logical" (we are not really defining a new type but using an alias for an existing one). Alias declarations are also more universal (i.e. cover more use cases) so if we want to standardize on one approach, it makes sense to use alias declarations.

> there appears to be no need for "using"

Yes, both `using` and `typedef` are legal in this context from C++ point of view.

> when the rest of Squid code has typedef

Since [2017](https://github.com/squid-cache/squid/commit/e99fa72) (at least), Squid uses both -- there are more than 20 non-templated alias declarations in master today. Thus, this code line does not introduce a new inconsistency. Moreover, when I find the time to work on it, one of the [modernization](http://lists.squid-cache.org/pipermail/squid-dev/2020-April/009569.html) changes might remove the remaining 1000 typedefs.

Reply:
Very well. If you are wanting to make a formal switch to remove typedef I agree.